core/src/appleMain/kotlin/Connection.kt

```diff
@@ -59,9 +62,4 @@ internal class Connection(
         if (error != null) throw IOException(error.description, cause = null)
         response as T
     }
-
-    fun close() {
```

Comment:
Is the idea now that instead of closing a Connection, you'll cancel its scope?

Reply:
From a consumer standpoint, nothing will change. `Peripheral.disconnect` will continue to disconnect. Ultimately, cancelling a `Peripheral`'s parent scope will also cancel (force dispose) the peripheral.

This is more of an internal book-keeping change. We no longer have to explicitly cancel the `Connection`'s `scope` because we now make its parent the scope of the `SharedRepeatableAction`: https://github.com/JuulLabs/kable/blob/e30220928290d52031ebcb3a17cf5e8cb449bd13/core/src/appleMain/kotlin/Peripheral.kt#L191 _(which will be cleaned up automatically on disconnect)_

`ShareRepeatableAction` compartmentalizes the connection lifecycle nicely: https://github.com/JuulLabs/kable/blob/e30220928290d52031ebcb3a17cf5e8cb449bd13/core/src/commonMain/kotlin/SharedRepeatableAction.kt